rules/msk_topic_config.go

```diff
@@ -0,0 +1,144 @@
+package rules
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/logger"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
+// MskTopicConfigRule checks the configuration for an MSK topic.
+type MskTopicConfigRule struct {
+	tflint.DefaultRule
+}
+
+func (r *MskTopicConfigRule) Name() string {
+	return "msk_topic_config"
+}
+
+func (r *MskTopicConfigRule) Enabled() bool {
+	return true
+}
+
+func (r *MskTopicConfigRule) Link() string {
+	return ReferenceLink(r.Name())
+}
+
+func (r *MskTopicConfigRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+func (r *MskTopicConfigRule) Check(runner tflint.Runner) error {
+	path, err := runner.GetModulePath()
+	if err != nil {
+		return fmt.Errorf("getting module path: %w", err)
+	}
+	if !path.IsRoot() {
+		logger.Debug("skipping child module")
+		return nil
+	}
+
+	resourceContents, err := runner.GetResourceContent(
+		"kafka_topic",
+		&hclext.BodySchema{
+			Attributes: []hclext.AttributeSchema{
+				{Name: "name"},
+				{Name: "replication_factor"},
+			},
+			Blocks: []hclext.BlockSchema{
+				{
+					Type: "config",
+					Body: &hclext.BodySchema{
+						Attributes: []hclext.AttributeSchema{
+							{Name: "retention.ms"},
+							{Name: "compression.type"},
+							{Name: "cleanup.policy"},
+						},
+					},
+				},
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return fmt.Errorf("getting kafka_topic contents: %w", err)
+	}
+
+	for _, topicResource := range resourceContents.Blocks {
+		if err := r.validateTopicConfig(runner, topicResource); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *MskTopicConfigRule) validateTopicConfig(runner tflint.Runner, topic *hclext.Block) error {
+	if err := r.validateReplicationFactor(runner, topic); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const (
+	replFactorAttrName   = "replication_factor"
+	replicationFactorVal = 3
```

Comment:
could you link to some docs explaining why this value

Reply:
Added in https://github.com/utilitywarehouse/tflint-ruleset-kafka-config/pull/12/commits/f78223246610da3b0c599389c05720d470de5b0a